.github/workflows/scripts/ping-codeowners-on-new-issue.sh

```diff
@@ -0,0 +1,81 @@
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+#
+
+set -euo pipefail
+
+if [[ -z "${ISSUE:-}" || -z "${TITLE:-}" || -z "${BODY:-}" || -z "${OPENER:-}" ]]; then
+  echo "Missing one of ISSUE, TITLE, BODY, or OPENER, please ensure all are set."
+  exit 0
+fi
+
+COMPONENT_REGEX='(cmd|pkg|extension|receiver|processor|exporter)'
+LABELS_COMMENT='See [Adding Labels via Comments](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-labels-via-comments) if you do not have permissions to add labels yourself.'
+CUR_DIRECTORY=$(dirname "$0")
+LABELS=""
+PING_LINES=""
+declare -A ADDED_LABELS
+
+TITLE_COMPONENT=$(echo "${TITLE}" | grep -oE "\[${COMPONENT_REGEX}/.+\]" | sed -E "s%^\[(${COMPONENT_REGEX}/.+)\]%\1%" || true)
+
+COMPONENTS_SECTION_START=$( (echo "${BODY}" | grep -n '### Component(s)' | awk '{ print $1 }' | grep -oE '[0-9]+') || echo '-1' )
+BODY_COMPONENTS=""
+
+if [[ "${COMPONENTS_SECTION_START}" != '-1' ]]; then
+  BODY_COMPONENTS=$(echo "${BODY}" | sed -n $((COMPONENTS_SECTION_START+2))p)
+fi
+
+if [[ -n "${TITLE_COMPONENT}" && ! ("${TITLE_COMPONENT}" =~ " ") ]]; then
+  # Transforms e.g. exporter/otlpexporter into exporter/otlp
+  TITLE_COMPONENT=$(echo "${TITLE_COMPONENT}" | sed -E "s%${COMPONENT_REGEX}/(.+)${COMPONENT_REGEX}%\1/\2%")
+  CODEOWNERS=$(COMPONENT="${TITLE_COMPONENT}" "${CUR_DIRECTORY}/get-codeowners.sh" || true)
+  
+  if [[ -n "${CODEOWNERS}" && ! ("${CODEOWNERS}" =~ ${OPENER}) ]]; then
+    ADDED_LABELS["${TITLE_COMPONENT}"]=1
+    LABELS+="${TITLE_COMPONENT}"
+    PING_LINES+="- ${TITLE_COMPONENT}: ${CODEOWNERS}\n"
+  fi
+fi
+
+for COMPONENT in ${BODY_COMPONENTS}; do
+  # Comments are delimited by ', ' and the for loop separates on spaces, so remove the extra comma.
+  COMPONENT=${COMPONENT//,/}
+  
+  CODEOWNERS=$(COMPONENT="${COMPONENT}" "${CUR_DIRECTORY}/get-codeowners.sh" || true)
+  
+  if [[ -n "${CODEOWNERS}" && ! ("${CODEOWNERS}" =~ ${OPENER}) ]]; then
+    if [[ -v ADDED_LABELS["${COMPONENT}"] ]]; then
+      continue
+    fi
+    if [[ -n "${LABELS}" ]]; then
+      LABELS+=","
+    fi
+    LABELS+="${COMPONENT}"
+    PING_LINES+="- ${COMPONENT}: ${CODEOWNERS}\n"
+  fi
+done
+
+if [[ -n "${LABELS}" ]]; then
+  # The GitHub CLI may fail if a tag doesn't exist, so stop this from failing the job.
+  gh issue edit "${ISSUE}" --add-label "${LABELS}" || true
+
+  # The GitHub CLI only offers multiline strings through file input.
+  # The PING_LINES variable must be directly put into the printf string
+  # to get the newlines to render correctly.
+  printf "Pinging code owners:\n${PING_LINES}\n%s" "${LABELS_COMMENT}"  \
+  | gh issue comment "${ISSUE}" -F -
```

Comment:
Do we need this part given that we already have hooks on labels creation? Won't this result in a duplicated comment?

Reply:
GitHub Actions disallows triggering workflows from other workflows to prevent loops. It knows who the actor is based on the auth token (user or bot), and in the case of bot actions, the event is not fired. Unfortunately this prevents what would otherwise be helpful composition of workflows, so we have to both apply the labels and ping the code owners here.

I've added a note to this call to help document its intent.

Reply:
Makes sense. Thanks for clarifying